albatross/covariance_functions/covariance_function.h

```diff
@@ -96,12 +89,21 @@ class CovarianceFunction : public ParameterHandlingMixin {
                 "\n\t\t..."
                 "\n\t}\n");
 
-  std::string get_name() const {
-    static_assert(has_name_<Derived>::value, "A public member `std::string "
-                                             "name_` must be defined for all "
-                                             "covariance functions");
-    return derived().name_;
-  };
+  template <typename DummyType = Derived,
+            typename std::enable_if<!has_name<DummyType>::value, int>::type = 0>
+  std::string get_name() {
+    static_assert(std::is_same<DummyType, Derived>::value,
+                  "never do covariance_function.get_name<T>()");
+    return typeid(Derived).name();
```

Comment:
Yeah I think I'm ok with that in this case since these methods really aren't used much, more for convenience when debugging.  If it concerns you having it in here though I'm happy to remove the default `get_name()` which uses `typeid` and force a user implementing a model to define `std::string name() const`.

Reply:
No reason to get rid of it. In the past I've usually disabled RTTI, but that's because the projects have been trying to shave off every last byte they could from the executable. I don't think we're anywhere near needing to shrink the executable size in Orion.